File: Java/프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.java
class Solution {
    public int[] solution(int n) {
        int[] answer = new int[n * (n + 1) / 2];
        int[][] tri = new int[n][n];
        int x = -1, y = 0, num = 1, index = 0;

        for (int i = 0; i < n; i++) {
            for (int j = i; j < n; j++) {
                if (i % 3 == 0) {
                    x++;
                } else if (i % 3 == 1) {
                    y++;
                } else {
                    x--;
                    y--;
                }
                tri[x][y] = num++;
            }
        }

        for (int[] arr : tri) {
            for (int i : arr) {
                if (i == 0) {
                    break;
                }
                answer[index++] = i;
            }
        }
        
        return answer;
    }
}

File: Python/프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.py
def solution(n):
    answer = [0] * ((n * (n + 1)) // 2)
    tri = [[0] * n for _ in range(n)]
    x, y, num, index = -1, 0, 1, 0

    for i in range(n):
        for j in range(i, n):
            if i % 3 == 0:
                x += 1
            elif i % 3 == 1:
                y += 1
            else:
                x -= 1
                y -= 1
            tri[x][y] = num
            num += 1

    for arr in tri:
        for i in arr:
            if i == 0:
                break
            answer[index] = i
            index += 1

    return answer